### PR TITLE
Stage 5 Restructure

### PR DESCRIPTION
**Pull request by StarWitch**
Restructured Stage 5 directory. No co-routinization done yet. I only changed a couple of function names for clarity/consistency/workability, but for the most part it's untouched.

### CODE REVIEW

**Review by Akaricchi — CHANGES_REQUESTED** [review]
LGTM for the most part!

File: src/stages/stage5/iku.c [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+	if(time == 0) {
+		// don't lerp; the spawn position is very different on hard/lunatic and would cause false hits
+		p->prevpos = p->pos;
+	}
+
+	p->angle = carg(p->args[0]*cexp(p->args[1]*t)*(1+p->args[1]*t));
+	return 1;
+}
+
```
[the remark this answers is not in the document]
Comment (StarWitch) [reply src/stages/stage5/iku.c]: `induction_bullet` (now `iku_induction_bullet`) is used in both `natural_cathode` and `induction`, so I'm keeping it in `iku.c` for now.

File: src/stages/stage5/nonspells/midboss_survival_1.c [hunk 2]
```diff
@@ -8,7 +8,7 @@
 
 #include "taisei.h"
 
-#include "nonspells.h"
+#include "survival.h"
```
Comment (Akaricchi) [inline src/stages/stage5/nonspells/midboss_survival_1.c]: Ditto. Just put it into `spells.h`; a separate header is overkill.

**Review by Akaricchi — COMMENTED** [review]
Sorry, minor pedantic detail

**Comment by StarWitch** [comment]
Done and done. Also rebased off of master to resolve the conflicts.

**Review by Akaricchi — APPROVED** [review]

**Pull request merged**